usb/esp_tinyusb/tusb_msc_storage.c

```diff
@@ -248,13 +249,24 @@ static esp_err_t _mount(char *drv, FATFS *fs)
 
 esp_err_t tinyusb_msc_storage_mount(const char *base_path)
 {
-    esp_err_t ret;
+    esp_err_t ret = ESP_OK;
     assert(s_storage_handle);
 
     if (s_storage_handle->is_fat_mounted) {
         return ESP_OK;
     }
 
+    tusb_msc_callback_t cb = s_storage_handle->callback_premount_changed;
+    if (cb) {
+        tinyusb_msc_event_t event = {
+            .type = TINYUSB_MSC_EVENT_PREMOUNT_CHANGED,
+            .mount_changed_data = {
+                .is_mounted = s_storage_handle->is_fat_mounted
```

Comment:
Nitpicking: if this code runs single threaded, 'is_fat_mounted' can only equal to FALSE here (similar situation few lines below). Please, consider either the code readability upgrade or adding some explanatory comments throughout whole component. It would surely help the users (incl. myself) with better understanding the logic. Not blocker now, though, can be done in follow-up PR. Thanks

Reply:
`is_fat_mounted` can have two values TRUE and FALSE.
The pre-mount/unmount callback can be delivered to users from two places - one from `tinyusb_msc_storage_mount()` before mount operation where `is_fat_mounted` is always FALSE and other from `tinyusb_msc_storage_unmount()` before unmount operation where `is_fat_mounted` is always TRUE. This is well intended. User can figure out the place from where the callback is generated based on the value of `is_fat_mounted`.

Similar explanation for post-mount/unmount callback.

I will add some code comments for users/developers.